src/editors/EditorContainer2.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useEffect } from 'react';
+import EditorPortal from './EditorPortal';
+import { CalculatedColumn } from '../common/types';
+import { InteractionMasksProps } from '../masks/InteractionMasks';
+import { UpdateActions } from '../common/enums';
+
+interface EditorContainerProps<R, K extends keyof R> extends Pick<InteractionMasksProps<R, K>, 'canvasRef' | 'rowHeight' | 'scrollTop' | 'scrollLeft' | 'onRowsUpdate'> {
+  column: CalculatedColumn<R>;
+  row: R;
+  rowIdx: number;
+  onClose(): void;
+}
+
+export default function EditorContainer<R, K extends keyof R>({
+  column,
+  row,
+  rowIdx,
+  rowHeight,
+  canvasRef,
+  scrollTop,
+  scrollLeft,
+  onRowsUpdate,
+  onClose
+}: EditorContainerProps<R, K>) {
+  const Editor = column.editor2!;
+
+  // close editor when scrolling
+  useEffect(() => onClose, [onClose, scrollTop, scrollLeft]);
+
+  function onChange<K extends keyof R>(rowUpdate: { [k in K]: R[k] }) {
+    onRowsUpdate({
+      cellKey: column.key,
+      fromRow: rowIdx,
+      toRow: rowIdx,
+      updated: rowUpdate as never, // TODO: fix type, or data structure
+      action: UpdateActions.CELL_UPDATE
+    });
+  }
+
+  function getContainerStyle(): React.CSSProperties | undefined {
+    if (!canvasRef.current) return;
+    const { top, left } = canvasRef.current.getBoundingClientRect();
```

Comment:
Let's fix this bug in canary in case someone use it? Should we also not check editorPortal
https://github.com/adazzle/react-data-grid/pull/1871/files#diff-63a962509ad28f9bf4dbc6c8a1c50569L153